src/CodeSigning/CodeSigning/Commands/InvokeCIPolicySigning.cs

```diff
@@ -0,0 +1,179 @@
+    {
+        #region Parameter Set Names
+
+        private const string ByAccountProfileNameParameterSet = "ByAccountProfileNameParameterSet";
+        private const string ByMetadataFileParameterSet = "ByMetadataFileParameterSet";
+        /// <summary>
+        /// The root element inside an SI Policy XML file.
+        /// </summary>
+        private static readonly XName SiPolicyRootElementName = XName.Get("SiPolicy", "urn:schemas-microsoft-com:sipolicy");
+
+        #endregion
+
+        #region Input Parameter Definitions
+
+        /// <summary>
+        /// Account Profile name
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The account name of Azure CodeSigning.")]
+        [ValidateNotNullOrEmpty]
+        public string AccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 1,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The certificate profile name of Azure CodeSigning account.")]
+        [ValidateNotNullOrEmpty()]
+        public string ProfileName { get; set; }
+        [Parameter(Mandatory = true,
+            Position = 2,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The endpoint url used to submit request to Azure CodeSigning.")]
+        public string EndpointUrl { get; set; }
+
+
+        /// <summary>
+        /// Metadata File Path
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ByMetadataFileParameterSet,
+             ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Metadata File path.")]
+        [ValidateNotNullOrEmpty]
+        public string MetadataFilePath { get; set; }
+
+        //common parameters
+        [Parameter(Mandatory = true,
+           Position = 3,
+           ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Original unsigned CI policy file path.")]
+        [Parameter(Mandatory = true,
+           Position = 3,
+           ParameterSetName = ByMetadataFileParameterSet,
+            ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Original unsigned CI policy file path.")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
+        [Parameter(Mandatory = true,
+           Position = 4,
+           ParameterSetName = ByAccountProfileNameParameterSet,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Signed CI policy file path")]
+        [Parameter(Mandatory = true,
+           Position = 4,
```

Comment:
Position 1 and position 2 are missed in parameter set `ByMetadataFileParameterSet`. And we do not suggest use position more than 3

Reply:
I fixed the order for the set highlighted, good catch. Concerning the suggestion, do need that bunch of parameters for invoking, any suggestion or guidance on viable alternatives?

Reply:
Removed the Position 4 and 5 from ci policy signing